modules/metastorage/src/test/java/org/apache/ignite/internal/metastorage/WatchAggregatorTest.java

```diff
@@ -131,6 +147,167 @@ public void testCancelByFalseFromListener() {
 
     }
 
+    /**
+     *
+     */
+    @Test
+    public void testOneCriterionInference() {
+        var watchAggregator = new WatchAggregator();
+
+        watchAggregator.add(new ByteArray("key"), null);
+
+        var keyCriterion = watchAggregator.watch(0, null).get().keyCriterion();
+        assertEquals(new KeyCriterion.ExactCriterion(new ByteArray("key")), keyCriterion);
```

Comment:
Please try to add a human-readable message that provides details.
"Test failed: expected 1, actual 2" looks very concise to me.